app/models/search_builder.rb

```diff
@@ -51,6 +52,11 @@ def run_advanced_parse?(solr_parameters)
       cleaned_query(solr_parameters[:q]) == solr_parameters[:q]
   end
 
+  def facets_for_advanced_search_form(solr_p)
```

Comment:
No, this is just for the solr query used to generate all the facet values and counts for the advanced search screen.  I'm really glad you asked, though, especially since the version I had earlier got all the wrong counts -- here's an updated version.